### PR TITLE
Extend tools to handle multiple files

### PR DESCRIPTION
**Pull request by gh-devnull**
Extend DiffTool, MergeTool, EditTool, and ShowTool to support one or more files in each tool.

### CODE REVIEW

**Comment by Murmele** [comment]
@gh-devnull can you format? In the repository there is a helper script cl-fmt.sh which formats the code with clang-format

We will probably release next 1 or two weeks, I have to check if I find the time to review this one. Otherwise it will land directly after the release.

**Comment by gh-devnull** [comment]
On 4/16/23 05:01, Murmele wrote:
>
> @gh-devnull <https://github.com/gh-devnull> can you format? In the 
> repository there is a helper script cl-fmt.sh which formats the code with 
> clang-format
>
@Murmele: Done.

File: src/tools/DiffTool.cpp [hunk 1]
```diff
@@ -35,76 +41,92 @@ bool DiffTool::start() {
...
-  });
+  int numFiles = mFiles.size(), i = -1;
+  foreach (const QString &file, mFiles) {
+    ++i;
+    git::Blob fileBlob;
+    if (!DiffTool::getBlob(file, git::Diff::OldFile, fileBlob))
+      continue;
+
+    // Get the path to any file blob.
+    QString blobMoniker;
```
[the remark this answers is not in the document]
Comment (gh-devnull) [reply src/tools/DiffTool.cpp:52]: My apologies.  This is a rather obscure name in somewhat buggy code. ;-)

I managed to mess up this branch with later updates so I removed it, closed the PR, and replaced them.  The new PR is #535.  The intent and most of the code is the same but I noticed that I broke DiffTool if using a commit or a stash and that the MergeTool wasn't using the workspace file as it should.  Both of these issues are resolved in PR #535.

Again, apologies for the confusion.  As an added bonus, there is no "blobMoniker" in #535. :-)

[the remark this answers is not in the document]
Comment (Murmele) [reply src/tools/DiffTool.cpp:52]: Haha I thought already why you closed the pr 😄 thanks I will check it out next days so we can get it in

**Pull request closed without merging**